frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts

```diff
@@ -505,3 +652,36 @@ export const createSinkConnection = (
 ): Promise<K8sResourceKind> => {
   return createTopologySinkConnection(sourceNode.getData(), targetNode.getData());
 };
+
+export const createEventingPubSubSink = (subObj: K8sResourceKind, target: K8sResourceKind) => {
+  if (!subObj || !target) {
+    return Promise.reject();
+  }
+  const subscriptionObj = _.omit(subObj, 'status');
+  const sink = {
+    ref: {
+      apiVersion: target.apiVersion,
+      kind: target.kind,
+      name: target.metadata?.name,
+    },
+  };
+  const updatePayload = {
+    ...subscriptionObj,
+    spec: { ...subscriptionObj.spec, subscriber: { ...sink } },
+  };
+
+  return k8sUpdate(modelFor(referenceFor(subscriptionObj)), updatePayload);
+};
+
+export const createSinkPubSubConnection = (
+  connector: any,
```

Comment:
can we pass sourceNode here and type it as `Node` and extract resources from the sourceNode?

Reply:
This util will enable change of subscriber from one knSvc to another through drag and drop or action menus so used connector resource. As connector here is subscription and it has all the info i.e channel and subscriber. Source node is channel that doesn't have these relationships and one channel can go to multiple subscriptions so even if we add it in resources object, still would need connector in context to map.